addons/mail/static/src/web/chatter.js

```diff
@@ -272,10 +272,14 @@ export class Chatter extends Component {
     }
 
     async onClickFollow() {
-        await this.orm.call(this.props.threadModel, "message_subscribe", [[this.props.threadId]], {
-            partner_ids: [this.store.self.id],
-        });
-        this.onFollowerChanged();
+        if (this.props.threadId) {
+            await this.orm.call(this.props.threadModel, "message_subscribe", [[this.props.threadId]], {
+                partner_ids: [this.store.self.id],
+            });
+            this.onFollowerChanged();
+        } else {
+            await this.props.saveRecord?.();
+        }
```

Comment:
```suggestion
        await this.props.saveRecord?.();
        await this.orm.call(this.props.threadModel, "message_subscribe", [[this.props.threadId]], {
            partner_ids: [this.store.self.id],
        });
        this.onFollowerChanged();
```

So this just becomes a single-line fix  `await this.props.saveRecord?.();`

Reply:
Hello @alexkuhn 

Yes, it became a single-line fix, but it did not work see https://drive.google.com/file/d/19Xw1KB46DPQ9_awWM1jDovPXGG9Csi9F/view?usp=drive_link.




In the above video, when we click new on a project and click on follow, it creates a follower record with 'project.project' id '0' with the current user and shows the required field. But when we click a second or more time, it shows a validation error and shows that the user already follows this project ( Key (res_model, res_id, partner_id) = (project.project, 0, 3) already exists).

In 'project.task, the traceback will still be generated.

In my POV: I think we can make follow disable when record is not save means not thread id by using `isDisabled`
https://github.com/odoo/odoo/blob/0901b6efb26b6662af607523a270d8b022d6b6ac/addons/mail/static/src/web/chatter.xml#L35
https://github.com/odoo/odoo/blob/0901b6efb26b6662af607523a270d8b022d6b6ac/addons/mail/static/src/web/chatter.js#L187-L189

Reply:
Can do this to handle when saving fails:
```js
const saved = await this.props.saveRecord?.();
if (!saved) {
    return;
}
```

The buttons were disabled when in draft before 16.0, but since 16.0 there's no longer edit / readonly mode on form view, therefore we decided (with FP) to keep these buttons working and silently attempt saving record if we can.
The reasoning for not disabling the buttons is because when people make a trial of Odoo in 16.0, they thought these buttons were not working, and it gave a bad 1st impression on the software.